File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    let left = "";

    for (let i = 1; i < food.length; i++) {
        let count = Math.floor(food[i] / 2); // 공정성을 위해 양쪽으로 나눌 수 있는 수
        left += String(i).repeat(count);     // 왼쪽에 배치
    }

    const right = [...left].reverse().join(""); // 오른쪽은 좌측 반대 순서
    return left + "0" + right; // 중앙에 물 배치
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    left = ""

    for i in range(1, len(food)):
        count = food[i] // 2  # 양쪽에 공평하게 나눌 수 있는 개수
        left += str(i) * count

    right = left[::-1]  # 왼쪽을 뒤집은 문자열
    answer = left + "0" + right

    return answer
